.github/workflows/dev-pytest.yml

```diff
@@ -0,0 +1,32 @@
+name: dev-pytest
+
+on:
+  push:
+    branches:
+      - dev
+
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
```

Comment:
For dev, let's pick one version of Python for now. This should speed up tests where only simple changes are occurring. 

```
    steps:
      - uses: actions/checkout@v4
      - name: Set up Python 3.11
        uses: actions/setup-python@v4
        with:
          python-version: "3.11"
```